edgar_tool/url_generator.py

```diff
@@ -1,94 +1,108 @@
 import datetime
-from typing import Literal, TypedDict
+from typing import Optional, TypedDict
 from urllib import parse
 
-from edgar_tool.constants import PEO_IN_AND_INC_IN_TO_SEC_FORM_ID, TEXT_SEARCH_BASE_URL
+import pydantic
 
+from edgar_tool import constants
 
-class SearchQueryKwargs(TypedDict, total=False):
-    keywords: list[str]
-    entity: str
-    filing_category: str
-    single_forms: list[str]
-    date_range_select: Literal["all", "10y", "1y", "30d", "custom"]
-    start_date: datetime.date
-    end_date: datetime.date
-    inc_in: str
-    peo_in: str
 
+class _SearchQueryKwargsValidator(pydantic.BaseModel):
+    keywords: Optional[list[str]] = None
+    entity: Optional[str] = None
+    filing_category: Optional[constants.FilingCategoryLiteral] = None
+    single_forms: Optional[list[str]] = None
+    date_range_select: Optional[constants.DateRangeLiteral] = None
+    start_date: Optional[datetime.date] = None
+    end_date: Optional[datetime.date] = None
+    inc_in: Optional[constants.LocationLiteral] = None
+    peo_in: Optional[constants.LocationLiteral] = None
```

Comment:
I need to update these to allow for multiple locations. The current CLI makes use of the `locationCodes` query argument when there are multiple.

Reply:
Done in https://github.com/bellingcat/EDGAR/pull/50/commits/2d8d8a6504d7437fa4e1c2895e363dabf249ff93